lib/connection.ex

```diff
@@ -291,6 +291,68 @@ defmodule Connection do
   """
   defcallback terminate(any, any) :: any
 
+  defmacro __using__(_) do
+    quote location: :keep do
+      @behaviour Connection
+
+      # The default implementations of init/1, handle_call/3, handle_info/2,
+      # handle_cast/2, terminate/2 and code_change/3 have been taken verbatim
+      # from Elixir's GenServer default implementation.
+
+      @doc false
+      def init(args) do
+        {:ok, args}
```

Comment:
`{:ok, args}` is not a valid return for `init/1` in `Connection`. Initially I wanted the behaviour to be a superset of GenServer. However it is important to know whether the process is going to connect (or not) so I decided it should be explicit about what the first step is after `init/1`. The callback returns are:
```elixir
 defcallback init(any) ::
    {:connect, any, any} |
    {:backoff, timeout, any} | {:backoff, timeout, any, timeout | :hibernate} |
    {:noconnect, any} | {:noconnect, any, timeout | :hibernate} |
    :ignore | {:stop, any}
```
What is a good default?